### ISSUE TITLE
Update the Quartz integration to v3 APIs

### ISSUE THREAD

**Issue by tafs7** [swimtver/Topshelf.StructureMap#7]
Quartz.NET v3.0 updated several APIs to support async/await.

This integration package does not support the newer 3.0 API.  Would love to upgrade my Quartz.NET Nuget package, but that would break this integration package at runtime (which is even worse, as we don't get any compiler errors during build).

Here's one of a few places that definitely breaks:
https://github.com/swimtver/Topshelf.StructureMap/blob/26e79144632a131b593eba10f33c7def70084202/src/Topshelf.Quartz.Structuremap/StructureMapScheduleConfiguratorExtensions.cs#L42

**Comment by tbd-develop** [comment]
I checked out the code and made a change to get it working, although I'm not sure it's the correct thing to do. Basically, the problem is IScheduler is now Task<IScheduler> throughout the Quartz API and we're expecting a IScheduler. Rather than concern myself with the whole Async design (not quite sure how Async comes into DI cleanly), I just tacked on .Result where we need it. Testing in the sample project works fine. Also took the opportunity to update the libraries to 4.7.1.

I would love to understand the right answer here (and the harm in adding .Result), but in the meantime, I have a solution that works for me.
